app/controllers/api/users.php

```diff
@@ -107,43 +110,60 @@
     ->label('sdk.response.code', Response::STATUS_CODE_OK)
     ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
     ->label('sdk.response.model', Response::MODEL_USER_LIST)
+    ->param('queries', [], new ArrayList(new Text(APP_LIMIT_ARRAY_ELEMENT_SIZE), APP_LIMIT_ARRAY_PARAMS_SIZE), 'Array of query strings generated using the Query class provided by the SDK. [Learn more about queries](https://appwrite.io/docs/databases#querying-documents). Maximum of ' . APP_LIMIT_ARRAY_PARAMS_SIZE . ' queries are allowed, each ' . APP_LIMIT_ARRAY_ELEMENT_SIZE . ' characters long. You may filter on the following attributes: ' . implode(', ', ListUsers::ALLOWED_ATTRIBUTES), true)
```

Comment:
We should use the new extended queries array in this stage unless absolutely not possible.

Reply:
Sorry, what do you mean?

Reply:
```suggestion
    ->param('queries', [], new Users(), 'Array of query strings generated using the Query class provided by the SDK. [Learn more about queries](https://appwrite.io/docs/databases#querying-documents). Maximum of ' . APP_LIMIT_ARRAY_PARAMS_SIZE . ' queries are allowed, each ' . APP_LIMIT_ARRAY_ELEMENT_SIZE . ' characters long. You may filter on the following attributes: ' . implode(', ', ListUsers::ALLOWED_ATTRIBUTES), true)
```

Reply:
Something like this 👆

Reply:
The queries need to be parsed before the data can be passed to the validator. If I put the pre-processing into the validator, wouldn't I need to parse the queries again in the body?

Reply:
We now have these validation classes. Resolving discussion.